### PR TITLE
Polkadot update v0.9.42

### LINKED ISSUE #313: upgrade dependencies to polkadot-0.9.42
* not sure if we need to upgrade substrate-fixed too

make sure to merge first:
* #308 (while not strictly necessary, runtime migration can only get trickier the longer we wait)

### PR DESCRIPTION
**Pull request by OverOrion**
Closes #313.

### CODE REVIEW

File: ceremonies/src/lib.rs [hunk 1]
```diff
@@ -67,11 +67,8 @@ pub mod pallet {
 	use frame_support::pallet_prelude::*;
 	use frame_system::pallet_prelude::*;
 
-	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
```
Comment (clangenb) [inline ceremonies/src/lib.rs:70]: Why did you remove this? We need this as we implemented storage migration lately.

File: communities/Cargo.toml [hunk 2]
```diff
@@ -15,20 +17,22 @@ encointer-primitives = { path = "../primitives", default-features = false }
...
 
 [dev-dependencies]
 approx = "0.5.1"
-sp-core = { version = "7.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.37" }
+sp-core = { version = "7.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
 test-utils = { path = "../test-utils" }
 
 [features]
+
+
```
Comment (clangenb) [inline communities/Cargo.toml:34]: please fix these lines

File: communities/src/lib.rs [hunk 3]
```diff
@@ -54,11 +54,8 @@ pub mod pallet {
 	use frame_support::pallet_prelude::*;
 	use frame_system::pallet_prelude::*;
 
-	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
```
Comment (clangenb) [inline communities/src/lib.rs:57]: Same here, I think we need this. @pifragile am I correct?

File: primitives/src/sybil.rs [hunk 4]
```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2019 Alain Brenzikofer
```
Comment (clangenb) [inline primitives/src/sybil.rs:1]: This file has been removed on master and is probably a rebase error

File: communities/src/lib.rs (hunk 3 above)
Comment (clangenb) [reply communities/src/lib.rs:57]: Ahh, is this maybe also just a rebase error?

**Review by clangenb — COMMENTED** [review]
Nice, looks in general very good thanks for the hard work!

I just saw at least one rebase error, which I would like you to fix. :)

**Comment by clangenb** [comment]
This pr will also close #301

File: communities/src/lib.rs (hunk 3 above)
Comment (OverOrion) [reply communities/src/lib.rs:57]: Yeah, I lost track of what's where during multiple merge conflicts, sorry about this! :sweat_smile:

File: primitives/src/sybil.rs (hunk 4 above)
Comment (OverOrion) [reply primitives/src/sybil.rs]: Yep, fixed in 5cd2060.

File: communities/src/lib.rs (hunk 3 above)
Comment (OverOrion) [reply communities/src/lib.rs:57]: Fixed in [4dc5183](https://github.com/encointer/pallets/pull/322/commits/4dc518340aadcdb64148393ff27ac6ee526df993).

File: ceremonies/src/lib.rs (hunk 1 above)
Comment (OverOrion) [reply ceremonies/src/lib.rs:70]: Fixed it [4dc5183](https://github.com/encointer/pallets/pull/322/commits/4dc518340aadcdb64148393ff27ac6ee526df993).

File: communities/Cargo.toml (hunk 2 above)
Comment (OverOrion) [reply communities/Cargo.toml]: Fixed in 7600955.

File: ceremonies/src/lib.rs [hunk 5]
```diff
@@ -68,10 +68,9 @@ pub mod pallet {
 	use frame_system::pallet_prelude::*;
 
 	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
-
 	#[pallet::pallet]
 	#[pallet::storage_version(STORAGE_VERSION)]
-	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::without_storage_info]
 	pub struct Pallet<T>(PhantomData<T>);
```
Comment (clangenb) [inline ceremonies/src/lib.rs:74]: This change is also part of the bad rebase if I am not mistaken.

File: communities/src/lib.rs [hunk 6]
```diff
@@ -55,10 +55,8 @@ pub mod pallet {
 	use frame_system::pallet_prelude::*;
 
 	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
-
 	#[pallet::pallet]
 	#[pallet::storage_version(STORAGE_VERSION)]
-	#[pallet::generate_store(pub (super) trait Store)]
```
Comment (clangenb) [inline communities/src/lib.rs:61]: Same here, generate store is needed, although I can't tell you all its semantics :P

File: faucet/src/lib.rs [hunk 7]
```diff
@@ -59,7 +59,6 @@ pub mod pallet {
 	use frame_system::pallet_prelude::*;
 
 	#[pallet::pallet]
-	#[pallet::generate_store(pub(super) trait Store)]
```
Comment (clangenb) [inline faucet/src/lib.rs:62]: I will stop commenting on this, please add it everywhere. :)

**Review by clangenb — COMMENTED** [review]
Nice thanks for all the hard work here. It looks good to me now except for one type of inconsistency!

File: communities/src/lib.rs (hunk 6 above)
Comment (clangenb) [reply communities/src/lib.rs:61]: Beautiful, thanks for the comment. I was sure this is not the case. XD
